react-hot-button-component/src/index.jsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+
+class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { clickCount: 0 };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    let count = this.state.clickCount;
+    count++;
+    this.setState({ clickCount: count });
+  }
+
+  render() {
+    const count = this.state.clickCount;
+    if (count <= 3) {
+      return <button onClick={this.handleClick} className='btn'>Hot Button</button>;
+    } else if (count <= 6) {
+      return <button onClick={this.handleClick} className='btn btn-cool'>Hot Button</button>;
+    } else if (count <= 9) {
+      return <button onClick={this.handleClick} className='btn btn-neutral'>Hot Button</button>;
+    } else if (count <= 12) {
+      return <button onClick={this.handleClick} className='btn btn-warm'>Hot Button</button>;
+    } else if (count <= 15) {
+      return <button onClick={this.handleClick} className='btn btn-hot'>Hot Button</button>;
+    } else if (count <= 18) {
+      return <button onClick={this.handleClick} className='btn btn-fuego'>Hot Button</button>;
+    }
```

Comment:
This is a lot of repetition and would be hard to maintain. A better approach would be to figure out the classname then return one button at the end.

```JSX
const count = this.state.clickCount;
let btnClass = '';

if (count <= 6) {
  btnClass = 'btn-cool';
} else if (count <= 9) {
  btnClass = 'btn-neutral';
} else if (count <= 12) {
  btnClass = 'btn-warm';
} else if (count <= 15) {
  btnClass = 'btn-hot';
} else if (count <= 18) {
  btnClass ='btn-fuego';
}

return <button onClick={this.handleClick} className={`btn ${btnClass}`}>Hot Button</button>;
```